### ISSUE TITLE
Nan loss

### ISSUE THREAD

**Issue by DennisCraandijk** [cvignac/SMP#3]
Hi there. First, thanks a lot for releasing the code of your great paper!

While trying the model on my own node classification task I keep getting a nan loss, preventing the model from learning. I can see from your multi-task training loop that you've encountered this behaviour as well and automatically stop training after some nan_counts. I've tried to trace back the origin of the nan value's, but don't really understand where they arise. 

Could you explain which operations cause the nan values? Are they a problem and can they be mitigated?

Thanks!

**Comment by cvignac** [comment]
Hello Dennis,
NaNs usually arise when the activations explode. There may be several reasons for that. The two that come to my mind are:
 - The learning rate is too big
 - There are too many layers and no normalization after each layer

There are several things that you can try:
  - reduce the learning rate
  - reduce the value of gradient clipping, which is used to avoid exploding parameters
  - use batch normalization, or another kind of normalization
  - reduce the number of layers

I indeed encountered a few NaNs at some point, but I think that gradient clipping solved the issue.
